### ISSUE TITLE
get uri for environment test: retrieve service connect DNS names failing

### ISSUE THREAD

**Issue by bijucyborg** [aws/copilot-cli#5225]
Hi,

I'm facing a strange problem

I'm deploying a simple manifest

name: redis
type: Backend Service

```
image:
  build: 
    dockerfile: ./Dockerfile.redis
    context: ./aiverify/setup-aiverify/aiverify-user
  healthcheck:
    command: ["CMD-SHELL", "redis-cli ping || exit 1"]
    interval: 10s
    retries: 2
    timeout: 5s
    start_period: 0s
  port: 6379

cpu: 1024       # Number of CPU units for the task.
memory: 2048    # Amount of memory in MiB used by the task.
count:
  spot: 1 # Number of tasks that should be running in your service.
exec: true     # Enable running commands in your container.
network:
  connect: true # Enable Service Connect for intra-environment traffic between services.

storage:
  volumes:
    appverifydata:
      path: /tempdata
      read_only: false
      efs:
        id: fs-0d38ecda017065663

```
At the end of the deployment, I observe the error, see the last line of the ouput below.

```
bijucyborg:~/environment $ copilot svc deploy --name redis --env test
WARNING! Your password will be stored unencrypted in /home/ec2-user/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
Building your container image "redis": docker build -t 416585528871.dkr.ecr.us-west-2.amazonaws.com/playground/redis:latest --label com.aws.copilot.image.builder=copilot-cli --label com.aws.copilot.image.container.name=redis --label com.aws.copilot.image.version=v1.29.1 /home/ec2-user/environment/aiverify/setup-aiverify/aiverify-user -f /home/ec2-user/environment/Dockerfile.redis
     6ff74796a0c8: Pushed
     8cbf51febd50: Pushed
     4693057ce236: Pushed
     latest: digest: sha256:a69cf3b0a8d2c6dede4fef5009f499357ac6a48ddf26887ffaa2fef742d15ffc size: 1778
     
✔ Proposing infrastructure changes for stack playground-test-redis
- Creating the infrastructure for stack playground-test-redis                 [create complete]  [126.8s]
  - Service discovery for your services to communicate within the VPC         [create complete]  [0.0s]
  - Update your environment's shared resources                                [create complete]  [2.9s]
  - An IAM role to update your environment stack                              [create complete]  [19.2s]
  - An IAM Role for the Fargate agent to make AWS API calls on your behalf    [create complete]  [15.9s]
  - A CloudWatch log group to hold your service logs                          [create complete]  [2.9s]
  - An ECS service to run and maintain your tasks in the environment cluster  [create complete]  [95.1s]
    Deployments                                                                                   
               Revision  Rollout      Desired  Running  Failed  Pending                                   
      PRIMARY  7         [completed]  1        1        0       0                                         
  - An ECS task definition to group your containers and run them on ECS       [create complete]  [0.0s]
  - An IAM role to control permissions for the containers in your tasks       [create complete]  [19.2s]
✔ Deployed service redis.
✘ get uri for environment test: retrieve service connect DNS names: get service redis: check if services are active: service "arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-YZbuyhk8ULyR/playground-test-redis-Service-wPfZQ8rHzFOd" and service "arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-71oRApKNdfbY/playground-test-redis-Service-wo2Fbpg5p8p1" should be in the same cluster
```

So I check the two services copilot is suggesting to me and one is the active cluster and the other is the inactive one.

```
bijucyborg:~/environment $ aws ecs describe-services --cluster playground-test-Cluster-YZbuyhk8ULyR --services arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-YZbuyhk8ULyR/playground-test-redis-Service-wPfZQ8rHzFOd
{
    "services": [
        {
            "networkConfiguration": {
                "awsvpcConfiguration": {
                    "subnets": [
                        "subnet-0ca594931ed16458a", 
                        "subnet-05ac70fa55dce5054"
                    ], 
                    "securityGroups": [
                        "sg-0b78f2f7364502284"
                    ], 
                    "assignPublicIp": "ENABLED"
                }
            }, 
            "enableECSManagedTags": false, 
            "loadBalancers": [], 
            "deploymentController": {
                "type": "ECS"
            }, 
            "desiredCount": 1, 
            "platformVersion": "LATEST", 
            "clusterArn": "arn:aws:ecs:us-west-2:416585528871:cluster/playground-test-Cluster-YZbuyhk8ULyR", 
            "serviceArn": "arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-YZbuyhk8ULyR/playground-test-redis-Service-wPfZQ8rHzFOd", 
            "deploymentConfiguration": {
                "deploymentCircuitBreaker": {
                    "enable": true, 
                    "rollback": true
                }, 
                "maximumPercent": 200, 
                "minimumHealthyPercent": 100
            }, 
            "createdAt": 1692978424.208, 
            "schedulingStrategy": "REPLICA", 
            "placementConstraints": [], 
            "deployments": [
                {
                    "status": "PRIMARY", 
                    "networkConfiguration": {
                        "awsvpcConfiguration": {
                            "subnets": [
                                "subnet-0ca594931ed16458a", 
                                "subnet-05ac70fa55dce5054"
                            ], 
                            "securityGroups": [
                                "sg-0b78f2f7364502284"
                            ], 
                            "assignPublicIp": "ENABLED"
                        }
                    }, 
                    "pendingCount": 0, 
                    "rolloutState": "COMPLETED", 
                    "rolloutStateReason": "ECS deployment ecs-svc/1661311021209418870 completed.", 
                    "createdAt": 1692978424.208, 
                    "desiredCount": 1, 
                    "platformVersion": "1.4.0", 
                    "failedTasks": 0, 
                    "taskDefinition": "arn:aws:ecs:us-west-2:416585528871:task-definition/playground-test-redis:7", 
                    "updatedAt": 1692978566.692, 
                    "capacityProviderStrategy": [
                        {
                            "capacityProvider": "FARGATE_SPOT", 
                            "base": 0, 
                            "weight": 1
                        }
                    ], 
                    "id": "ecs-svc/1661311021209418870", 
                    "runningCount": 1
                }
            ], 
            "serviceName": "playground-test-redis-Service-wPfZQ8rHzFOd", 
            "events": [
                {
                    "message": "(service playground-test-redis-Service-wPfZQ8rHzFOd) has reached a steady state.", 
                    "id": "7113ba71-cfc7-47d9-b6cf-97f37ffdc802", 
                    "createdAt": 1692978566.697
                }, 
                {
                    "message": "(service playground-test-redis-Service-wPfZQ8rHzFOd) has reached a steady state.", 
                    "id": "20067b58-4fc5-49e9-b9e6-c594a751e834", 
                    "createdAt": 1692978504.862
                }, 
                {
                    "message": "(service playground-test-redis-Service-wPfZQ8rHzFOd) (deployment ecs-svc/1661311021209418870) deployment completed.", 
                    "id": "39dd8460-50bd-48e2-bba1-a8d32365b2ef", 
                    "createdAt": 1692978504.861
                }, 
                {
                    "message": "(service playground-test-redis-Service-wPfZQ8rHzFOd) has started 1 tasks: (task dd23b621a065492dac53225eda4cc640).", 
                    "id": "369d2c03-1116-41a9-8bce-b155a7b79e84", 
                    "createdAt": 1692978426.977
                }
            ], 
            "runningCount": 1, 
            "status": "ACTIVE", 
            "serviceRegistries": [
                {
                    "registryArn": "arn:aws:servicediscovery:us-west-2:416585528871:service/srv-l4lh65nz4wewotfy", 
                    "port": 6379
                }
            ], 
            "pendingCount": 0, 
            "createdBy": "arn:aws:iam::416585528871:role/playground-test-CFNExecutionRole", 
            "capacityProviderStrategy": [
                {
                    "capacityProvider": "FARGATE_SPOT", 
                    "base": 0, 
                    "weight": 1
                }
            ], 
            "placementStrategy": [], 
            "propagateTags": "SERVICE", 
            "roleArn": "arn:aws:iam::416585528871:role/aws-service-role/ecs.amazonaws.com/AWSServiceRoleForECS", 
            "taskDefinition": "arn:aws:ecs:us-west-2:416585528871:task-definition/playground-test-redis:7", 
            "enableExecuteCommand": true
        }
    ], 
    "failures": []
}

```
```
bijucyborg:~/environment $ aws ecs describe-services --cluster playground-test-Cluster-71oRApKNdfbY --services arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-71oRApKNdfbY/playground-test-redis-Service-wo2Fbpg5p8p1
{
    "services": [
        {
            "networkConfiguration": {
                "awsvpcConfiguration": {
                    "subnets": [
                        "subnet-0b4b93a9a5565abcc", 
                        "subnet-0a6a52d289bbaad79"
                    ], 
                    "securityGroups": [
                        "sg-0af294b89e2414d85"
                    ], 
                    "assignPublicIp": "ENABLED"
                }
            }, 
            "enableECSManagedTags": false, 
            "loadBalancers": [], 
            "deploymentController": {
                "type": "ECS"
            }, 
            "desiredCount": 0, 
            "platformVersion": "LATEST", 
            "clusterArn": "arn:aws:ecs:us-west-2:416585528871:cluster/playground-test-Cluster-71oRApKNdfbY", 
            "serviceArn": "arn:aws:ecs:us-west-2:416585528871:service/playground-test-Cluster-71oRApKNdfbY/playground-test-redis-Service-wo2Fbpg5p8p1", 
            "deploymentConfiguration": {
                "deploymentCircuitBreaker": {
                    "enable": true, 
                    "rollback": true
                }, 
                "maximumPercent": 200, 
                "minimumHealthyPercent": 100
            }, 
            "createdAt": 1692976140.404, 
            "schedulingStrategy": "REPLICA", 
            "placementConstraints": [], 
            "deployments": [
                {
                    "status": "PRIMARY", 
                    "networkConfiguration": {
                        "awsvpcConfiguration": {
                            "subnets": [
                                "subnet-0b4b93a9a5565abcc", 
                                "subnet-0a6a52d289bbaad79"
                            ], 
                            "securityGroups": [
                                "sg-0af294b89e2414d85"
                            ], 
                            "assignPublicIp": "ENABLED"
                        }
                    }, 
                    "pendingCount": 0, 
                    "rolloutState": "COMPLETED", 
                    "rolloutStateReason": "ECS deployment ecs-svc/3192313829564560316 completed.", 
                    "createdAt": 1692976140.404, 
                    "desiredCount": 0, 
                    "platformVersion": "1.4.0", 
                    "failedTasks": 0, 
                    "taskDefinition": "arn:aws:ecs:us-west-2:416585528871:task-definition/playground-test-redis:6", 
                    "updatedAt": 1692977023.105, 
                    "capacityProviderStrategy": [
                        {
                            "capacityProvider": "FARGATE_SPOT", 
                            "base": 0, 
                            "weight": 1
                        }
                    ], 
                    "id": "ecs-svc/3192313829564560316", 
                    "runningCount": 0
                }
            ], 
            "serviceName": "playground-test-redis-Service-wo2Fbpg5p8p1", 
            "events": [], 
            "runningCount": 0, 
            "status": "INACTIVE", 
            "serviceRegistries": [
                {
                    "registryArn": "arn:aws:servicediscovery:us-west-2:416585528871:service/srv-l4lh65nz4wewotfy", 
                    "port": 6379
                }
            ], 
            "pendingCount": 0, 
            "createdBy": "arn:aws:iam::416585528871:role/playground-test-CFNExecutionRole", 
            "capacityProviderStrategy": [
                {
                    "capacityProvider": "FARGATE_SPOT", 
                    "base": 0, 
                    "weight": 1
                }
            ], 
            "placementStrategy": [], 
            "propagateTags": "SERVICE", 
            "roleArn": "arn:aws:iam::416585528871:role/aws-service-role/ecs.amazonaws.com/AWSServiceRoleForECS", 
            "taskDefinition": "arn:aws:ecs:us-west-2:416585528871:task-definition/playground-test-redis:6", 
            "enableExecuteCommand": true
        }
    ], 
    "failures": []
}
```

Looks like there is some remanant object somewhere, although the cloudformation UI looks pretty clean.

![image](https://github.com/aws/copilot-cli/assets/24872459/c875219f-bf78-460d-a5fb-d20271e744bf)

**Comment by iamhopaul123** [comment]
Hello @bijucyborg. Sorry for the confusion. I think it should be fixed by https://github.com/aws/copilot-cli/commit/77c88c237af66b815648b671332278ba7d5cf120. And we'll have a release soon!

**Comment by dannyrandall** [comment]
Hi @bijucyborg - this fix was released in [v1.30.0](https://github.com/aws/copilot-cli/releases/tag/v1.30.0)!:rocket: Thanks for reporting the bug!
